### ISSUE TITLE
Cant build ghcjs using nix

### ISSUE THREAD

**Issue by mckeankylej** [commercialhaskell/stack#2298]
When trying to build ghcjs using the nix options, I get the following error:  

    <command line>: can't load .so/.DLL for: libncursesw.so (libncursesw.so: cannot open shared object file: No such file or directory)

In my stack.yaml:

```yaml
resolver: lts-5.8
compiler: ghcjs-0.2.0.20160414_ghc-7.10.3
compiler-check: match-exact
setup-info:
  ghcjs:
    source:
      ghcjs-0.2.0.20160414_ghc-7.10.3:
        url: https://s3.amazonaws.com/ghcjs/ghcjs-0.2.0.20160414_ghc-7.10.3.tar.gz
        sha1: 6d6f307503be9e94e0c96ef1308c7cf224d06be3

nix:
  enable: true
  packages: [ncurses,nodejs]

# Local packages, usually specified by relative directory name
packages:
  - location:
      git:    'https://github.com/ghcjs/ghcjs-base'
      commit: 'e9e9e36afde694202de034905efdeef153e1b04d'
    extra-dep: true

# Packages to be pulled from upstream that are not in the resolver (e.g., acme-missiles-0.3)
extra-deps:
  - 'ghcjs-dom-0.2.3.1'
  - 'jsaddle-0.3.0.3'
  - 'reflex-0.4.0'
  - 'reflex-dom-0.3'
  - 'ref-tf-0.4'
  - 'haskell-src-exts-1.16.0.1'
  - 'syb-0.5.1'
  - 'reflex-dom-helpers-0.1.0.0'
  - 'operational-extra-0.4'

# Override default flag values for local packages and extra-deps
flags: {}

# Extra package databases containing global packages
extra-package-dbs: []
```

I have the ncurses dependency in the packages section.

**Comment by seanparsons** [comment]
I've just seen this (or a variant thereof as I was trying to use ghcjs), I wonder if it's connected to this: https://github.com/commercialhaskell/stack/issues/440

**Comment by mckeankylej** [comment]
They said that problem should be fixed with ghc 7.10.3. Also I don't know how to apply that using nix. On top of all of this basically the entirety of nix with stack is broken for me [see here.](https://github.com/NixOS/nixpkgs/issues/16371)

**Comment by mckeankylej** [comment]
When #16371 was fixed this was also fixed.
